app/controllers/catalog_controller.rb

```diff
@@ -41,6 +42,48 @@ def update_user_per_cat_page
     end
   end
 
+  def submit_form # rubocop:disable MethodLength, AbcSize
+    flash.clear
+    begin
+      # set the start and end date to updated values
+      # expects date in format MM/DD/YYYY or MM/DD/YY
+      start_date = Date.strptime(params[:form][:start_date], '%m/%d/%Y')
+      due_date = Date.strptime(params[:form][:due_date], '%m/%d/%Y')
+      # Wcan't change the past
+      unless start_date >= Date.today && due_date >= Date.today
+        fail ArgumentError
+      end
+      cart.start_date = start_date
+      cart.due_date = due_date
+      cart.fix_due_date
+      flash[:notice] = 'Reservation updated.'
+    rescue ArgumentError
+      flash[:alert] = 'Please enter a valid start or due date.'
+    end
+
+    # get soft blackout notices
+    notices = []
+    notices << Blackout.get_notices_for_date(cart.start_date, :soft)
+    notices << Blackout.get_notices_for_date(cart.due_date, :soft)
+    notices = notices.reject(&:blank?).to_sentence
+    notices += "\n" unless notices.blank?
+    # update all the items in the cart
+    (0..(params[:quantity].count - 1)).each do |i|
+      @quantity = params[:quantity].values[i].to_i
+      @model_id = params[:quantity].keys[i].to_i
+      # make sure the quantity changed before looking for Models
+      if cart.items[@model_id] != @quantity
+        @equipment_model = EquipmentModel.find(@model_id)
```

Comment:
I don't think these need to be class variables (preceded by the `@` sign), they can just be regular variables (see [here](http://www.rubyist.net/~slagell/ruby/variables.html)).